Test-№2.0-from-23.11.2022/task1.c

```diff
@@ -0,0 +1,250 @@
+        {
+            return false;
+        }
+    }
+    for (int i = 0; i < comparesLostCnt; ++i)
+    {
+        switch (i)
+        {
+            case 0:
+                answer *= isItANumber(queue, i);
+                break;
+            case 1:
+                answer *= isItANumber(queue, i);
+                break;
+            case 2:
+                answer *= isItADot(queue, i);
+                break;
+            case 3:
+                answer *= isItANumber(queue, i);
+                break;
+            case 4:
+                answer *= isItANumber(queue, i);
+                break;
+            case 5:
+                answer *= isItADot(queue, i);
+                break;
+            case 6:
+                answer *= isItANumber(queue, i);
+                break;
+            case 7:
+                answer *= isItANumber(queue, i);
+                break;
+            case 8:
+                answer *= isItANumber(queue, i);
+                break;
+            case 9:
+                answer *= isItANumber(queue, i);
+                break;
+            default:
+                answer = false;
+        }
+        if (answer == false)
+        {
+            return false;
+        }
+    }
+    return true;
+}
+
+int fromCharToInt(char number)
+{
+    char numbers[10] = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9'};
+    for (int i = 0; i < 10; ++i)
+    {
+        if (number == numbers[i])
+        {
+            return i;
+        }
+    }
+    return -1;
+}
+
+void helpDateInit(char *helpDate, char *date)
+{
+    //date has this format: n n . n n . n n n n
+    //.mask for i for help: 0 1 2 3 4 5 6 7 8 9
+    helpDate[0] = date[6];
+    helpDate[1] = date[7];
+    helpDate[2] = date[8];
+    helpDate[3] = date[9];
+    helpDate[4] = date[3];
+    helpDate[5] = date[4];
+    helpDate[6] = date[0];
+    helpDate[7] = date[1];
+}
+
+bool isItABestDate(char *firstDate, char *secondDate)
+{
+    // The question is: Is it a firstDate are greater than secondDate?
+    // return 1 = yes
+    // return 0 = dates are equal
+    // return -1 = no
+    //date has this format: n n . n n . n n n n
+    //.mask for i for help: 0 1 2 3 4 5 6 7 8 9
+    char helpFirstDate[dateSize];
+    helpDateInit(helpFirstDate, firstDate);
+    char helpSecondDate[dateSize];
+    helpDateInit(helpSecondDate, secondDate);
+    int endOfIter = dateSize - 2;
+    for (int i = 0; i < endOfIter; ++i)
+    {
+        if (fromCharToInt(helpFirstDate[i]) > fromCharToInt(helpSecondDate[i]))
+        {
+            return true;
+        }
+        else if (fromCharToInt(helpFirstDate[i]) < fromCharToInt(helpSecondDate[i]))
+        {
+            return false;
+        }
+    }
+    return false;
+}
+
+void copyDate(char *from, char *to)
+{
+    for (int i = 0; i < dateSize; ++i)
+    {
+        to[i] = from[i];
+    }
+}
+
+void task1(FILE *file, char *answer, int *errorCode)
+{
+    bool flag = true;
+    char queue[dateSize] = {' '};

```

Comment:
```suggestion
    char dateBuffer[dateSize] = {' '};
```
`buffer` - стандартное название для массива, в котором накапливается значение.